### DISCUSSION TITLE
Passing OpenXML objects into ClosedXML?

### DISCUSSION THREAD

**Discussion by mojofawad** [ClosedXML/ClosedXML#2214] [Q&A] [answered]
I'm trying to incorporate ClosedXML into our app that currently uses OpenXML. 

A short version of my issue is that this doesn't work

This:
```csharp
using (SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Open(filepath, true))
{
        using (var workbook = new XLWorkbook(filepath)) // <-- breaks here
        {
            // some logic
        }
}
```

Returns:
```
System.IO.IOException: The process cannot access the file 'F:ile\path\here' because it is being used by another process.
```

What does work is changing isEditable to false like so - `SpreadsheetDocument.Open(filepath, false))`. In the short term, I can't make that change in our application. Long term, we're considering switching over entirely. 

My workaround currently is to create a temporary copy to open when utilizing ClosedXML. However, if there's a way to pass OpenXML objects directly into ClosedXML then I would greatly appreciate being pointed in the right direction

**Comment by jahav** [comment, marked as answer]
There is no way to do that. OpenXML Sdk code/objects is used only during loading and saving and the long term plan is to remove OpenXML SDK as a dependency.

**Reply by mojofawad** [reply]
Ah no worries! Thanks for letting me know. My solution for now is to create a copy before opening up the document with OpenXML and then opening the copy with ClosedXML since I just need ClosedXML for reading.
